gnat2goto/driver/range_check.adb

```diff
@@ -144,20 +164,160 @@ package body Range_Check is
       end case;
    end Get_Bound_Of_Bounded_Type;
 
-   ----------------------------
-   -- Make_Range_Assert_Expr --
-   ----------------------------
+   function Make_Div_Zero_Assert_Expr (N : Node_Id; Value : Irep;
+                                       Divisor : Irep) return Irep
+   is
+      Value_Type : constant Irep :=
+        Follow_Symbol_Type (Get_Type (Value), Global_Symbol_Table);
+      Source_Loc : constant Source_Ptr := Sloc (N);
+
+      function Build_Assert_Function return Symbol;
+
+      ---------------------------
+      -- Build_Assert_Function --
+      ---------------------------
+
+      --  Build a symbol for the following function
+      --  Value_Type division_check(Value_Type value, Value_Type divisor)
+      --  {
+      --    assert (divisor != 0);
+      --    return value;
```

Comment:
A couple of things : 1. the default behaviour of all of the check functions should be `assert(condition); assume(condition)`, 2. why pass `Value_Type`?  It should be enough to take a single Boolean?

Reply:
Ad 1. The assert-then-assume behaviour is there, just the comment was wrong -> Updated. Ad 2. the division (and other) check may be call inside expression, thus it needs to return the result value of the division.

Reply:
OK... this may complicate what the "ignore" option is for these.